### ISSUE TITLE
Make multiclass_prefabs.c to present multiclasses for the Demo 

### ISSUE THREAD

**Issue by katie-gao** [uchicago-cs/chiventure#1672]
Jake, Van, and I are creating a new `multiclass_prefabs.c` to include the newly revamped multiclass module for the demo. Though this is primarily for the demo, it could also serve as a greater part of Chiventure as a whole. We will be creating combination prefab classes based on the single classes in `class_prefabs.c`. We will include the Hexblade (combination of Warrior and Wizard) that is already in the demo as one of these combinations, as well as several others.

**Comment by vanolson** [comment]
We've created `multiclass_prefabs.h` and `multiclass_prefabs.c`, with comments fleshed out on the .h file. We will now turn to implementing these functions.

**Comment by wellingtj** [comment]
The implementation is going well - where we now have "hexblade" (wizard, warrior), "infernal" (elementalist, sorcerer), "paladin" (knight, monk), "enchanter" (bard, healer), and "assassin" (rogue, ranger). There are two main functions 'multiclass_prefab_new' and 'multiclass_prefab_add_skills'. 'multiclass_prefab_new' will read the name and return the short desc, long desc, and set the stats using the revamped functions from 'multiclass.c'. This function works for all the new multiclasses. 'multiclass_prefab_add_skills' uses the skilltrees functions from 'multiclass.c' and prints this out in the demo. Now, we are just working on the starter skills.

**Comment by katie-gao** [comment]
We created both multiclass_prefabs.c and multiclass_prefabs.h for the playerclass demo. This was successful in actually printing out all the skills of multiclasses, alongside stats and damages. As Jake mentioned earlier, there are now four functional multiclass prefabs of "hexblade" (wizard, warrior), "infernal" (elementalist, sorcerer), "paladin" (knight, monk), "enchanter" (bard, healer), and "assassin" (rogue, ranger). While Van and I worked mainly on the compilation, multiclass_prefabs.h, and some helpers for multicalss_prefabs.c, Jake worked on the 'multiclass_prefab_new' function where it takes in the game and multiclass name and returns a multiclass. The biggest struggle was adding skilltrees, where the multiclass functions did not seem to work fully. However, we had silly errors, and the code ended up working. We also had helper functions 'get_multiclass_name_index' and default multiclass name array to store the prefabs. That way we could change all the names to lowercase, and have an easy time matching the name inputted to one of the prefabs. Now, that we have completed these functions and tested them with all the multiclass prefabs - we are closing this issue.
